packages/drivers/routerlicious-driver/src/urlUtils.ts

```diff
@@ -17,3 +19,23 @@ export const parseFluidUrl = (fluidUrl: string): URLParse => {
  */
 export const replaceDocumentIdInPath = (urlPath: string, documentId: string): string =>
     urlPath.split("/").slice(0, -1).concat([documentId]).join("/");
+
+export const createFluidUrl = (domain: string, pathname: string): string =>
+    "fluid://".concat(domain).concat(pathname);
+
+export const replaceWithDiscoveryUrl = (resolvedUrl: IFluidResolvedUrl,
+    session: ISession,
```

Comment:
Do we want to consider making IFluidResolvedUrl immutable and let replaceWithDiscoveryUrl return a new instance?

Reply:
Got it. Could I know what's the benefit of doing this? :) 

Also, resolveUrl consists of the other attribute. We might create a new instance by deep copy the resolveUrl object, and changing the ordererUrl, storageUrl, and deltaStorageUrl of it.

Reply:
Generally, immutability carries multiple advantages. This article discusses immutability in the context of Java, but the general ideas apply in other environments. [https://www.leadingagile.com/2018/03/immutable-in-java/#:~:text=Immutable%20objects%20offer%20a%20number,when%20we%20define%20mutable%20objects](https://www.leadingagile.com/2018/03/immutable-in-java/#:~:text=Immutable%20objects%20offer%20a%20number,when%20we%20define%20mutable%20objects).